### ISSUE TITLE
Entities invoked with the DispatchAsync don’t log full exceptions

### ISSUE THREAD

**Issue by jeffhollan** [Azure/azure-functions-durable-extension#862]
Created an entity function and tried to invoke via the REST API.  I failed to include the `op` query parameter and got this error:

> [7/16/19 10:00:18 PM] @Circuit@test: Function 'Circuit (Entity)' failed '' operation f138b510-86e2-4ee4-a0b0-4429bd0ff21f after 313.1957ms with exception (5488 bytes). Input: (296 bytes). IsReplay: False. HubName: debug. AppName: . SlotName: . ExtensionVersion: 2.0.0. SequenceNumber: 7.

Took me a bit to realize it was because I didn’t include the operation.  Would be good if error was something more like `No operation specified.  Please include the name of the operation you are trying to invoke with the “op” query parameter`

**Comment by cgillum** [comment]
One possible problem with adding this validation is that I know we do allow operations to have empty operation names (e.g. the "default" operation). In that case, I expect we would want to allow not specifying the `op` parameter, but I can see how this points to a broader discoverability problem. @sebastianburckhardt any thoughts on this?

**Comment by jeffhollan** [comment]
It actually may just be a broader issue that I don’t see the actual exception details being bubbled up.  I had another issue with invoking my actor when I did include operation and got similar error but can’t see the underlying exception details (what’s in all those bytes?)

> [7/16/19 10:21:22 PM] @Circuit@test: Function 'Circuit (Entity)' failed 'AddFailure' operation a37ea5bc-e3d6-484c-a496-cc73fe171dca after 230.8707ms with exception (12172 bytes). Input: (296 bytes). IsReplay: False. HubName: debug. AppName: . SlotName: . ExtensionVersion: 2.0.0. SequenceNumber: 3.

**Comment by sebastianburckhardt** [comment]
Yes, perhaps exceptions are not being made visible enough in general.

**Comment by jeffhollan** [comment]
Yeah renaming this to focus on exceptions in general.  Any exception that happens during my Durable Entity (via the `DispatchAsync` method) I don’t see bubbled up in my logs.  Just hit one where I tried to add a duplicate value to a dictionary and all I see is exception with 9332 bytes was hit (but no indication of the exception type or stack trace).

**Comment by sebastianburckhardt** [comment]
Exceptions thrown by operations are now more thoroughly surfaced (#946), so it should be a bit easier for users to figure out what's happening.

However, if anything goes wrong with an operation (whether it simply missing the op argument, using an incorrect op argument, or having wrong content in the body) the user still has to look at the portal (or the debug console, for local development) to see the error message. We currently have no easy way to surface that error in the http response - by definition we are not waiting for the signal to be delivered before returning the http response.

In principle, we could do slightly better for typed entities, by validating the operation name and format of the body. The problem is that with the current way we do typed entities, we don't actually know the type of an entity in the client... because this information becomes apparent only once the DispatchAsync is called.

**Comment by cgillum** [comment]
Closing now that improvements have been made via the linked PR.
